cargo/lib/dependabot/cargo/file_parser.rb

```diff
@@ -171,17 +171,20 @@ def registry_source_details(declaration)
                 "cargo config"
         end
 
-        # Use known values from crates.microsoft.com rather than
-        # querying them, which avoids having to handle authentication
-        # and the unusual "sparse+..." URL.
-        if index_url == "sparse+https://crates.microsoft.com/index/"
+        # "Handling" sparse registries in common::lib::Dependabot::Source is not 
+        # impossible (i.e., by getting the config.json file from the index endpoint) 
+        # but sparse registries have their indices at the supplied URI minus the 
+        # `sparse+` header, and we only need to call on the index URI - so just 
+        # supply that information here instead.
+        if index_url.start_with? "sparse+"
           {
-            type: "registry",
+            type: "registry+sparse",
             name: registry_name,
-            index: index_url,
-            dl: "https://crates.microsoft.com/api/v1/crates",
-            api: "https://crates.microsoft.com"
+            index: index_url[7..-2], # Chop off the final /.
```

Comment:
Ruby 2.5.5 added `delete_{suffix|prefix}(!)` which would be a lot nicer than these two magic numbers.  OTOH I don't know what version of Ruby we need to support.

Regardless, we shouldn't blindly trim the last character (since it might not be a `/`) we should either check for the `/` or (maybe better?) parse the index as a URI which will handle the separators for us.

Reply:
https://apidock.com/ruby/v2_5_5/String/delete_suffix%21

Reply:
Our downstream is 2.7.0; I think it's probably fine and it would be nice to replace these magic numbers, so I'll change to use `delete_` methods.